### ISSUE TITLE
Ability to track/tag requests

### ISSUE THREAD

**Issue by javixeneize** [zaproxy/zaproxy#4002]
Hi

I am using ZAP as a proxy to test an application from different computers. I am facing the problem that i dont know which request is coming from which computer

A nice enhancement would be to assign a tracker or a tag to those requests (and create the relevant API method to retrieve it) to be able to identify (and filter) which requests are coming from which computer

**Comment by psiinon** [comment]
@javixeneize have you tried using tags? you can automatically tag requests based on regexes, and then filter on those tags in the History table.
Also related to #4000

**Comment by psiinon** [comment]
https://github.com/zaproxy/zap-core-help/wiki/HelpStartConceptsTags

**Comment by javixeneize** [comment]
But this is tagging the existing requests, it doesnt fit my scenario

I want to run traffic automatically though ZAP from selenium, from different computers, and later, automatically, i want to do a python script that retrieves those requests from the history

If the tag can be automatically added when running traffic automatically, it will solve my issue, but i dont think this parameter can be sent when sending traffic through ZAP, isnt it?

**Comment by kingthorin** [comment]
The thing that @psiinon linked tags traffic passively as it passes through the proxy.

Ref: https://groups.google.com/forum/#!topic/zaproxy-users/z1l2fkGoAVc

> 
> You could use a script to tag things, if you can differentiate things some how. For example: https://github.com/zaproxy/community-scripts/blob/master/standalone/historySourceTagger.js
> 
> There's also these issues that might be relevant:
> https://github.com/zaproxy/zaproxy/issues/3983
> https://github.com/zaproxy/zaproxy/issues/4000
> 
> If you really want one or both of those to move (or any issue really) you could vote it up on bountysource, or actually put a bounty on them.
> https://www.bountysource.com/teams/zap/issues
> 
> The last option is to run multiple copies of ZAP with different proxy config a and keep the two users' data separate.

**Comment by javixeneize** [comment]
I will raise it in bountysource. I think it is something that will help, specially when you are using a single point proxy for different computers

The script says "The script can be run multiple times, history entries will only be tagged". My point is... how can you assign a tag in execution time? Can this be called from selenium? It is, can selenium tag assign a tag automatically to those requests calling this js?

**Comment by thc202** [comment]
Updated the title of the issue to reflect better what's being requested.

**Comment by kingthorin** [comment]
> The script says "The script can be run multiple times, history entries will only be tagged". My point is... how can you assign a tag in execution time? Can this be called from selenium? It is, can selenium tag assign a tag automatically to those requests calling this js?

Yes the script is after the fact. Per the link @psiinon provided you can tag anything going through the proxy if you have a pattern that can be looked for. Selenium can use a proxy, therefore Selenium traffic can be tagged....

**Comment by kingthorin** [comment]
I believe this should be do'able with a proxy or HttpSender script.

**Comment by thc202** [comment]
The address is not available.

**Comment by kingthorin** [comment]
😢  okay, after I said it I wondered if the information was too buried or maybe not exposed at all, but hadn't had a chance to go poking around for it.
